go/enclave/core/rollup.go

```diff
@@ -38,26 +41,38 @@ func (r *Rollup) IsGenesis() bool {
 	return r.Header.Number.Cmp(big.NewInt(int64(common.L2GenesisHeight))) == 0
 }
 
-func (r *Rollup) ToExtRollup(txBlobCrypto crypto.TransactionBlobCrypto) *common.ExtRollup {
-	extBatches := make([]*common.ExtBatch, len(r.Batches))
-	for idx, batch := range r.Batches {
-		extBatches[idx] = batch.ToExtBatch(txBlobCrypto)
+func (r *Rollup) ToExtRollup(txBlobCrypto crypto.DataEncryptionService, compression crypto.DataCompressionService) (*common.ExtRollup, error) {
+	plaintextBatchesBlob, err := rlp.EncodeToBytes(r.Batches)
+	if err != nil {
+		return nil, err
+	}
+
+	compressedBatchesBlob, err := compression.Compress(plaintextBatchesBlob)
+	if err != nil {
+		return nil, err
 	}
 
 	return &common.ExtRollup{
 		Header:  r.Header,
-		Batches: extBatches,
-	}
+		Batches: txBlobCrypto.Encrypt(compressedBatchesBlob),
```

Comment:
Do we want to encrypt batches? This would also encrypt their headers

Reply:
Umm.
Do you think there could be another unencrypted by compressed payload only for headers?
Which could be used by tooling
Yeah, that sounds like a good idea